### PR TITLE
Add mokey12 keyboard

### PR DESCRIPTION
**Pull request by rhmokey**
## Description
Add mokey12 keyboard
A Blue Pill STM32F103C8T6-based keyboard.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mokey/mokey12/config.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+
+#define RGB_DI_PIN B15
+#define RGBLED_NUM 16
```
Comment (waffle87) [inline keyboards/mokey/mokey12/config.h:41]: This should all be migrated to `info.json`. See [doc](https://docs.qmk.fm/#/reference_info_json).
```suggestion
```

File: keyboards/mokey/mokey12/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+	),
+    [3] = LAYOUT(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
```
Comment (waffle87) [inline keyboards/mokey/mokey12/keymaps/via/keymap.c:43]:
```suggestion
```

File: keyboards/mokey/mokey12/mcuconf.h
Comment (waffle87) [inline keyboards/mokey/mokey12/mcuconf.h:1]: This file should be removed.

File: keyboards/mokey/mokey12/mokey12.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            }
+        }
+        uint16_t held_keycode_timer = timer_read();
+        register_code(mapped_code);
+        while (timer_elapsed(held_keycode_timer) < MEDIA_KEY_DELAY) { /* no-op */
+        }
+        unregister_code(mapped_code);
+    }
+    return true;
+}
```
Comment (waffle87) [inline keyboards/mokey/mokey12/mokey12.c:66]: This approach would be preferable, as the current implementation does not account for someone entering 16-bit keycodes or other things along those lines. 
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise) {
  if (!encoder_update_user(index, clockwise)) {
    return false;
  }
  switch (get_highest_layer(layer_state)) {
    case 0:
    default:
      if (clockwise) {
        tap_code_delay(KC_VOLU, 10);
      } else {
        tap_code_delay(KC_VOLD, 10);
      }
      break;
    case 1:
      if (clockwise) {
        tap_code_delay(KC_MNXT, 10);
      } else {
        tap_code_delay(KC_MPRV, 10);
      }
      break;
    case 2:
      if (clockwise) {
        tap_code_delay(KC_PGDN, 10);
      } else {
        tap_code_delay(KC_PGUP, 10);
      }
      break;
  }
  return true;
}
```

File: keyboards/mokey/mokey12/readme.md
Comment (waffle87) [inline keyboards/mokey/mokey12/readme.md:1]: This file's contents should follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md).

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/mokey/mokey12/rules.mk [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = yes
+SLEEP_LED_ENABLE = yes
```
[the remark this answers is not in the document]
Comment (rhmokey) [reply keyboards/mokey/mokey12/rules.mk:12]: # Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes	# Mouse keys
EXTRAKEY_ENABLE = yes	# Audio control and System control
CONSOLE_ENABLE = no	# Console for debug
COMMAND_ENABLE = no    # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no
RGBLIGHT_ENABLE = yes
SLEEP_LED_ENABLE = no
WS2812_DRIVER = spi
ENCODER_ENABLE = yes

# Enter lower-power sleep mode when on the ChibiOS idle thread
OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE

**Pull request closed without merging**